library/test/src/test/java/com/bumptech/glide/manager/PostMConnectivityMonitorTest.java

```diff
@@ -0,0 +1,100 @@
+package com.bumptech.glide.manager;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import android.content.Context;
+import android.net.ConnectivityManager;
+import android.net.ConnectivityManager.NetworkCallback;
+import android.net.Network;
+import android.net.NetworkCapabilities;
+import com.bumptech.glide.manager.PostMConnectivityMonitorTest.PermissionConnectivityManager;
+import java.util.Set;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+import org.robolectric.annotation.Config;
+import org.robolectric.annotation.Implementation;
+import org.robolectric.annotation.Implements;
+import org.robolectric.shadow.api.Shadow;
+import org.robolectric.shadows.ShadowConnectivityManager;
+
+@RunWith(RobolectricTestRunner.class)
+@Config(sdk = 23, shadows = PermissionConnectivityManager.class)
+public class PostMConnectivityMonitorTest {
+  @Mock private ConnectivityMonitor.ConnectivityListener listener;
+  private PostMConnectivityMonitor monitor;
+  private ConnectivityHarness harness;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    monitor = new PostMConnectivityMonitor(RuntimeEnvironment.application, listener);
+    harness = new ConnectivityHarness();
+  }
+
+  @Test
+  public void testRegistersNetworkCallbackOnRegister() {
+    monitor.register();
+    assertThat(harness.getNetworkCall()).contains(monitor.callback);
+  }
+
+  @Test
+  public void testDoesNotRegisterTwiceOnRegister() {
+    monitor.register();
+    monitor.register();
+    assertThat(harness.getNetworkCall()).contains(monitor.callback);
```

Comment:
This does not actually verify it's only registered once - I suggest you use `.containsExactly(monitor.callback)`

Reply:
ok